src/python/tests/core/bot/fuzzers/libFuzzer/libfuzzer_launcher_test.py

```diff
@@ -1804,6 +1817,348 @@ def test_fuzz_with_mutations_using_ml_rnn(self, mock_execute, *_):
+              u'strategy_recommended_dict':
+                  0,
+              u'strategy_value_profile':
+                  0,
+              u'timeout_count':
+                  0,
+              u'timeout_limit':
+                  25,
+              'timestamp':
+                  1337.0
+          })
+
+      # Output printed.
+      self.assertIn(self.no_crash_output, mock_stdout.getvalue())
+      expected_command = (
+          'Command: /fake/build_dir/fake_fuzzer '
+          '-max_len=80 -rss_limit_mb=2048 -timeout=25 '
+          '-collect_data_flow=/fake/dfsan_build/fake_fuzzer -fork=1 '
+          '-artifact_prefix=/fake/ -max_total_time=2650 -print_final_stats=1 '
+          '/fake/inputs-disk/temp-1337/new /fake/corpus_dir')
+      self.assertIn(expected_command, mock_stdout.getvalue())
+      self.assertIn('Bot: test-bot', mock_stdout.getvalue())
+      self.assertIn('Time ran:', mock_stdout.getvalue())
+
+  @mock.patch('bot.fuzzers.libFuzzer.launcher.add_recommended_dictionary',
+              lambda x, y, z: False)
+  @mock.patch('multiprocessing.cpu_count', return_value=1)
+  @mock.patch('sys.stdout', new_callable=test_utils.MockStdout)
+  @mock.patch('system.minijail.tempfile.NamedTemporaryFile')
+  def test_fuzz_with_dataflow_tracing_minijail(self, mock_tempfile, *_):
+    """Tests fuzzing with dataflow tracing inside minijail."""
+    self.mock.do_dataflow_tracing.return_value = True
+    os.environ['USE_MINIJAIL'] = 'True'
+
+    mock_tempfile.return_value.__enter__.return_value.name = '/tmppath'
+    mock_tempfile.return_value.name = '/tmpfile'
+
+    self.fs.CreateDirectory('/fake/corpus_dir')
+    self.fs.CreateFile('/fake/testcase_basic')
+    self.fs.CreateDirectory('/fake/dfsan_build')
+    self.fs.CreateFile('/fake/dfsan_build/fake_fuzzer')
+
+    os.environ['FUZZ_CORPUS_DIR'] = '/fake/corpus_dir'
+    os.environ['DATAFLOW_BUILD_DIR'] = '/fake/dfsan_build'
+
+    new_units_added = 11
+    with mock.patch(
+        'subprocess.Popen',
+        create_mock_popen(
+            self.no_crash_output, '/fake/inputs-disk/temp-1337/new',
+            '/fake/main_corpus_dir', new_units_added)) as mock_popen:
+      launcher.main([
+          'launcher.py',
+          '/fake/testcase_basic',
+          'fake_fuzzer',
+          '-max_len=80',
+      ])
+
+      self.assertEqual(mock_popen.commands, [[
+          'sudo',
+          '-S',
+          'mknod',
+          '-m',
+          '666',
+          '/fake/inputs-disk/temp-1337/CHROOT/dev/null',
+          'c',
+          '1',
+          '3',
+      ], [
+          'sudo', '-S', 'mknod', '-m', '666',
+          '/fake/inputs-disk/temp-1337/CHROOT/dev/random', 'c', '1', '8'
+      ], [
+          'sudo',
+          '-S',
+          'mknod',
+          '-m',
+          '666',
+          '/fake/inputs-disk/temp-1337/CHROOT/dev/urandom',
+          'c',
+          '1',
+          '9',
+      ], [
+          '/fake_root/resources/platform/{}/minijail0'.format(
```

Comment:
`os.path.join('/fake_root/resources/platform', environment.platform().lower(), `minijail0')`seems preferable to calling format here.

Reply:
Other tests in this file are using `'{}'.format`.

Reply:
` '{}'.format` is inconsistent with the rest of the codebase which uses `os.path.join`.
Please change this (and the others if you feel the inconsistency withing the file is important).

Reply:
It's a bikeshedding at this point. My opinion regarding consistency is based on my previous software engineering experience. Looks like Google guidelines (http://google.github.io/styleguide/pyguide.html#4-parting-words) are mentioning something along this lines, but of course that's not a discussion I'd really like to spend time on.

If we want to refactor something, it's potentially a good task and should be done thoroughly in a separate CL. I've filed https://github.com/google/clusterfuzz/issues/594 to make sure your comments aren't lost and we'll eventually address those or close as WontFix.

Please let me know if you think there are any issues that are crucial for enabling the dataflow tracing strategy support.